libraries/ui-library/src/components/six-input/six-input.tsx

```diff
@@ -205,45 +205,52 @@ export class SixInput {
     this.handleClearClick = this.handleClearClick.bind(this);
     this.handlePasswordToggle = this.handlePasswordToggle.bind(this);
     this.handleSlotChange = this.handleSlotChange.bind(this);
-
-    this.host.shadowRoot.addEventListener('slotchange', this.handleSlotChange);
+    this.host.shadowRoot?.addEventListener('slotchange', this.handleSlotChange);
   }
 
   componentWillLoad() {
-    this.defaultValue = this.value;
+    this.defaultValue = this.getValue();
```

Comment:
Is it guaranteed, that the "value" is set before the "componentWillLoad" callback is triggered? Otherwise the "defaultValue" will never be set.

Reply:
Yes, it should be, if the value is set as an attribute. But I wonder if we should ever set the `defaultValue` to the actual value. It could be misleading/counter-intuitive, that the first value for the input will be used as the default value. Other options I can think of options:

1.- Provide a property to set the default value explicitly, e.g. `<six-input value="100", defaultValue="0">`. (e.g. [shoelace ](https://shoelace.style/components/input ) seems to do it like this.
2. Deprecate/remove the reset/defaultValue functionality and let the users of the library implement it

My favorite is option 2.

Reply:
let's keep the default value for now for backwards compability